### DISCUSSION TITLE
The documentation lacks info about what the RefreshDatabase trait does

### DISCUSSION THREAD

**Discussion by SuperDJ** [laravel/framework#35025] [Q&A] [answered]
In my unit tests I need to use models which rely on different database connections. In the tests I'm using the `RefreshDatabase` trait. However I couldn't find any info in the documentation about what the trait does exactly.  What is documented about the trait is clear but I think it lacks info about multiple database connections specifically.

>It is often useful to reset your database after each test so that data from a previous test does not interfere with subsequent tests. The RefreshDatabase trait takes the most optimal approach to migrating your test database depending on if you are using an in-memory database or a traditional database. Use the trait on your test class and everything will be handled for you

- I mean is it refreshing all database connections or only the default connection?
- If it is refreshing all database connections is there a configuration setting to disable this behavior?

**Comment by paras-malhotra** [comment, marked as answer]
It is equivalent to the following:

1. `php artisan migrate:fresh` on your default connection (per your `phpunit.xml` with fallback to app default) +
2. Wrapping all DB calls in a transaction that's rolled back on tear down

**Comment by SuperDJ** [comment]
Created PR which will add this info to the docs https://github.com/laravel/docs/pull/6540
